Source/WebCore/loader/MixedContentChecker.cpp

```diff
@@ -184,11 +190,11 @@ bool MixedContentChecker::shouldUpgradeInsecureContent(LocalFrame& frame, IsUpgr
         // 4.1.2 request’s URL’s host is an IP address.
         || (!shouldUpgradeIPAddressAndLocalhostForTesting && URL::hostIsIPAddress(url.host()))
         // 4.1.4 request’s destination is not "image", "audio", or "video".
-        || (destination != FetchOptions::Destination::Audio && destination != FetchOptions::Destination::Image && destination != FetchOptions::Destination::Video)
+        || (!destinationIsImageAudioOrVideo(destination))
```

Comment:
Maybe we need to drop this condition, in favor of the one below (and move/adjust the comment)?

Reply:
I think we still want this, for cases where the mode is NOT cors. If just use the changes I propose below, we would decide not to check for image/audio/media if the mode was non-cors, and allow he load.